Evolutionary_loop/Loop_Parts/Part_B/Part_B_PUEO_job2.sh

```diff
@@ -0,0 +1,116 @@
+## I've commented these out because we needed to put them inside of a loop in the macroskeleton ##
+## Currently these are hardcoded outputmacroskeleton_GPU.xmacro ##
+#line1='var query = new ResultQuery();'
+#line2='///////////////////////Get Theta and Phi Gain///////////////'
+#line3='query.projectId = App.getActiveProject().getProjectDirectory();'
+freqlist="20000 21000 22000 23000 24000 25000 26000 27000 28000 29000 30000 31000 32000 33000 34000 35000 36000 37000 38000 39000 40000 41000 42000 43000 44000 45000 46000 47000 48000 49000 50000 51000 52000 53000 54000 55000 56000 57000 58000 59000 60000 61000 62000 63000 64000 65000 66000 67000 68000 69000 70000 71000 72000 73000 74000 75000 76000 77000 78000 79000 80000 81000 82000 83000 84000 85000 86000 87000 88000 89000 90000 91000 92000 93000 94000 95000 96000 97000 98000 99000 100000 101000 102000 103000 104000 105000 106000 107000 108000 109000 110000 111000 112000 113000 114000 115000 116000 117000 118000 119000 120000 121000 122000 123000 124000 125000 126000 127000 128000 129000 130000 131000 132000 133000 134000 135000 136000 137000 138000 139000 140000 141000 142000 143000 144000 145000 146000 147000 148000 149000 150000"
+#The list of frequencies, scaled up by 100 to avoid float operation errors in bash
+#we have to wait to change the frequencies since we're going to be changing them as we append them to simulation_PEC.xmacro (which is removed below before being remade)
+
+## We need the loop to pause until all the XF jobs are done
+## To do this, we'll just count the flag files
+
+cd $WorkingDir/Run_Outputs/$RunName/GPUFlags/
+flag_files=$(ls | wc -l) #$(ls -l --file-type | grep -v '/$' | wc -l)
+
+#Now we need to repeat that for the rest of the jobs
+while [[ $flag_files -lt $XFCOUNT ]] #we need to loop until flag_files reaches totPop
+do
+	sleep 1m
+	echo $flag_files
+	flag_files=$(ls | wc -l) #$(ls -l --file-type | grep -v '/$' | wc -l)
+done
+
+rm -f $WorkingDir/Run_Outputs/$RunName/GPUFlags/*
+
+echo $flag_files
+echo "Done!"
+
+# First, remove the old .xmacro files
+#when do that, we end up making the files only readable; we should just overwrite them
+#alternatively, we can just set them as rwe when the script makes them
+cd $XmacrosDir
+ 
+rm -f output.xmacro
+
+#echo "var m = $i;" >> output.xmacro
+echo "var NPOP = $NPOP;" >> output.xmacro
+echo "for (var k = $(($gen * $XFCOUNT + 1)); k <= $(($gen * $XFCOUNT + $XFCOUNT)); k++){" >> output.xmacro
+
+if [ $NSECTIONS -eq 1 ] # if 1, then the cone is symmetric
+then
+	cat shortened_outputmacroskeleton.txt >> output.xmacro
+else
+	cat shortened_outputmacroskeleton_Asym.txt >> output.xmacro
+fi
+
+
+sed -i "s+fileDirectory+${WorkingDir}+" output.xmacro
+# When we use the sed command, anything can be the delimiter between each of the arguments; usually, we use /, but since there are / in the thing we are trying to substitute in ($WorkingDir), we need to use a different delimiter that doesn't appear there
```

Comment:
```suggestion
# use | as delimiter for sed since WorkingDir contains /
sed -i "s|fileDirectory|${WorkingDir}|" output.xmacro
```
Before reading the comment and the expression more carefully I thought you were concatenating different vars to create a path.  I would suggest using a different delimiter (like `|`, which is a common alternative to `/`).  The comment can also simplified.

In addition, `fileDirectory` could be set as a variable at the beginning of the files, instead of using `sed` to replace it after the file has been written.  Using templates would be even better, but that's probably easier in Python than in bash.

Reply:
Comment changed for non-pueo files.
Delimiter changed or non-pueo files.

I'm not sure what you mean in the second paragraph. Do you mean that we can set fileDirectory as a variable inside the xmacro? 
I'm currently not too familiar with xmacro and the macroskeletons, but I will keep this in mind in the future.

Reply:
Sorry I'm still a little confused.
Yes, the .xmacro file is made from skeleton .txt files which contain the hardcoded word `fileDirectory` (by `cat`)
We then `sed` this word out once the .xmacro is made.
Are you saying that I should change the way the skeleton (.txt) files are written?

Reply:
That's the idea.  Even though I'm not too familiar with `.xmacro`, I would expect this to be doable, also because other variables are already set and written in the `.xmacro` file before copying the skeleton.

If `"fileDirectory"` is replaced with a variable, in the bash file you should be able to do something like
```sh
echo "var fileDirectory = ${WorkingDir};" >> simulation_PEC.xmacro
```
so that `fileDirectory` becomes a variable set at the top of `output.xmacro`, and in the skeleton replace
```js
var file = "fileDirectory/Antenna_Performance_Metric/";
```
with
```js
var file = fileDirectory + "/Antenna_Performance_Metric/";
```
so that the value of the `fileDirectory` variable is used instead of `"fileDirectory"` being an hardcoded string.

Reply:
Got it. I'll look into this but we should probably involve Alex if changes are to be made for the xmacro skeleton files, as they are used for PUEO as well. I've created a new issue for this so that I don't forget.